File: solutions/python/0013.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_map = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }
        result = 0
        for i in range(len(s)):
            current = roman_map[s[i]]
            next = roman_map[s[i + 1]] if i + 1 < len(s) else 0
            if next and current < next:
                result -= current
            else:
                result += current
        return result

File: solutions/rust/0013.rs
use std::collections::HashMap;

impl Solution {
    pub fn roman_to_int(s: String) -> i32 {
        let roman_map = HashMap::from([
            ('I', 1),
            ('V', 5),
            ('X', 10),
            ('L', 50),
            ('C', 100),
            ('D', 500),
            ('M', 1000)
        ]);
        let mut result = 0;
        for (i, c) in s.chars().enumerate() {
            let current = roman_map.get(&c).unwrap();
            let next = roman_map.get(&s.chars().nth(i + 1).unwrap_or(' ')).unwrap_or(&0);
            if current < next {
                result -= current;
            } else {
                result += current;
            }
        }
        result
    }
}
